web/client/src/entities/CallForwardSetting/Form.tsx

```diff
@@ -48,54 +48,28 @@ const Form = (props: EntityFormProps): JSX.Element => {
 
   const newProperties = { ...properties };
 
-  const targetType = { ...newProperties.targetType } as ScalarProperty;
-  const targetTypeEnum = { ...(targetType?.enum || {}) };
-
-  const callForwardType = { ...(newProperties?.callForwardType || {}) } as ScalarProperty;
-  const callForwardTypeEnum = { ...(callForwardType?.enum || {}) };
-
   const userPath = match.path.includes(User.path);
   const friendPath = match.path.includes(Friend.path);
   const retailAccountPath = match.path.includes(RetailAccount.path);
+  const residentialDevicePath = match.path.includes(ResidentialDevice.path);
 
-  if (userPath) {
-
-    delete newProperties.friend;
+  if (!retailAccountPath) {
     delete newProperties.cfwToRetailAccount;
-    delete newProperties.retailAccount;
-    delete newProperties.residentialDevice;
-
-  } else if (friendPath) {
+    delete newProperties.ddi;
+  }
 
-    delete newProperties.user;
-    delete newProperties.cfwToRetailAccount;
-    delete newProperties.retailAccount;
+  if (!residentialDevicePath) {
     delete newProperties.residentialDevice;
+  }
 
-  } else if (retailAccountPath) {
-
-    delete newProperties.friend;
+  if (!userPath) {
     delete newProperties.user;
-    delete newProperties.residentialDevice;
-
-    delete targetTypeEnum.extension;
-    delete targetTypeEnum.voicemail;
-
-    delete callForwardTypeEnum.noAnswer;
-    delete callForwardTypeEnum.busy;
-
-  } else if (match.path.includes(CallForwardSetting.path)) {
+  }
 
+  if (!friendPath) {
     delete newProperties.friend;
-    delete newProperties.user;
   }
 
-  targetType.enum = targetTypeEnum;
-  newProperties.targetType = targetType;
-
-  callForwardType.enum = callForwardTypeEnum;
-  newProperties.callForwardType = callForwardType;
-
```

Comment:
Only these fields seems pending:

 - **noAnswerTimeout**. This field is only used when the callForwardType is noAnswer, only supported in vPBX (user and friend) and Residential (residentialDevice)
 - **voicemail**. This field is only used when targetType is voicemail, only supported in vPBX (user and friend) and Residential (residentialDevice)
 - **extension**. This field is only used when targetType is extension, only supported in vPBX (user and friend) and Residential (residentialDevice)
 - **callTypeFilter**. This field is only used in vPBX (user and friend)

Reply:
I prefer to just let visualToggle handle their visibility